server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java

```diff
@@ -297,27 +303,20 @@ protected void produceMail(Map<String, Object> props, int msgPrio, Mail mail) th
         props.put(JAMES_MAIL_MESSAGE_SIZE, mail.getMessageSize());
         props.put(JAMES_MAIL_NAME, mail.getName());
 
-        StringBuilder recipientsBuilder = new StringBuilder();
-
-        Iterator<MailAddress> recipients = mail.getRecipients().iterator();
-        while (recipients.hasNext()) {
-            String recipient = recipients.next().toString();
-            recipientsBuilder.append(recipient.trim());
-            if (recipients.hasNext()) {
-                recipientsBuilder.append(JAMES_MAIL_SEPARATOR);
-            }
+        // won't serialize the empty headers so it is mandatory
```

Comment:
Why can't we serialize it when it's empty?

Reply:
The empty `PerRecipientHeaders` will cost us a `303` bytes and once we hexing it'll double it's size, so I think it'll way much better to not sent anything instead of pay additional bytes for just empty object. Generally speaking this little optimization intended to decrease number of bytes sent over the wire.

Reply:
Seems fine to me